components/remerge/src/schema/json.rs

```diff
@@ -0,0 +1,1178 @@
+        max: Option<f64>,
+
+        #[serde(default)]
+        #[serde(skip_serializing_if = "is_default")]
+        if_out_of_bounds: Option<IfOutOfBounds>,
+    },
+
+    #[serde(rename = "integer")]
+    Integer {
+        #[serde(flatten)]
+        common: RawFieldCommon<Option<i64>>,
+
+        #[serde(default)]
+        #[serde(skip_serializing_if = "is_default")]
+        min: Option<i64>,
+
+        #[serde(default)]
+        #[serde(skip_serializing_if = "is_default")]
+        max: Option<i64>,
+
+        #[serde(default)]
+        #[serde(skip_serializing_if = "is_default")]
+        if_out_of_bounds: Option<IfOutOfBounds>,
+    },
+
+    #[serde(rename = "timestamp")]
+    Timestamp {
+        #[serde(flatten)]
+        common: RawFieldCommon<Option<RawTimeDefault>>,
+
+        #[serde(default)]
+        #[serde(skip_serializing_if = "is_default")]
+        semantic: Option<RawTimestampSemantic>,
+    },
+
+    #[serde(rename = "own_guid")]
+    OwnGuid {
+        #[serde(flatten)]
+        // TODO: check that serde does what I want with the `()` field.
+        common: RawFieldCommon<()>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "is_default")]
+        auto: Option<bool>,
+    },
+
+    #[serde(rename = "untyped_map")]
+    UntypedMap {
+        #[serde(flatten)]
+        common: RawFieldCommon<Option<JsonObject>>,
+
+        #[serde(default)]
+        prefer_deletions: bool,
+    },
+
+    #[serde(rename = "record_set")]
+    RecordSet {
+        #[serde(flatten)]
+        common: RawFieldCommon<Option<Vec<JsonObject>>>,
+
+        // Note: required!
+        id_key: String,
+
+        #[serde(default)]
+        prefer_deletions: bool,
+    },
+}
+
+macro_rules! common_getter {
+    ($name:ident, $T:ty) => {
+        pub fn $name(&self) -> $T {
+            match self {
+                RawFieldType::Untyped { common, .. } => &common.$name,
+                RawFieldType::Text { common, .. } => &common.$name,
+                RawFieldType::Url { common, .. } => &common.$name,
+                RawFieldType::Boolean { common, .. } => &common.$name,
+                RawFieldType::Number { common, .. } => &common.$name,
+                RawFieldType::Integer { common, .. } => &common.$name,
+                RawFieldType::Timestamp { common, .. } => &common.$name,
+                RawFieldType::OwnGuid { common, .. } => &common.$name,
+                RawFieldType::RecordSet { common, .. } => &common.$name,
+                RawFieldType::UntypedMap { common, .. } => &common.$name,
+            }
+        }
+    };
+}
+
+impl RawFieldType {
+    common_getter!(name, &str);
+    common_getter!(local_name, &Option<String>);
+    common_getter!(required, &bool);
+    common_getter!(deprecated, &bool);
+    common_getter!(composite_root, &Option<String>);
+    common_getter!(merge, &Option<ParsedMerge>);
+    common_getter!(change_preference, &Option<ChangePreference>);
```

Comment:
Eek! I can't think of a better way to do this without changing the types around, to where `common` isn't generic, though, which is more trouble than it's worth.